validator/submain.py

```diff
@@ -386,6 +386,7 @@ def test_inner_package(err, xpi_package, for_appversions=None):
     if (err.get_resource('has_install_rdf') and
             not err.get_resource('has_manifest_json') and
             not err.get_resource('is_multiprocess_compatible') and
+            err.detected_type == PACKAGE_EXTENSION and
```

Comment:
`PACKAGE_MULTI` value is `1` just like `PACKAGE_EXTENSION` so it's already covered. We probably do need to include`PACKAGE_SUBPACKAGE`, though.